commit 767fca9e52ef
Author: Bobby Holley <bobbyholley@gmail.com>
Date:   Tue Jul 3 23:47:48 2018 +0000

    Bug 1431285 - Cap the number of style threads at six. r=emilio a=lizzard
    
    MozReview-Commit-ID: 3qI1mIvDn8j
    
    Differential Revision: https://phabricator.services.mozilla.com/D1928
    
    --HG--
    extra : source : 0d61bbcd24d8eb49d2beea278859da868186f783
---
 servo/components/style/gecko/global_style_data.rs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git servo/components/style/gecko/global_style_data.rs servo/components/style/gecko/global_style_data.rs
index f1cb107dff0fa..6671de9867d2f 100644
--- servo/components/style/gecko/global_style_data.rs
+++ servo/components/style/gecko/global_style_data.rs
@@ -65,7 +65,12 @@ lazy_static! {
             .map(|s| s.parse::<usize>().expect("invalid STYLO_THREADS value"));
         let mut num_threads = match stylo_threads {
             Ok(num) => num,
-            _ => cmp::max(num_cpus::get() * 3 / 4, 1),
+            // The default heuristic is num_virtual_cores * .75. This gives us
+            // three threads on a hyper-threaded dual core, and six threads on
+            // a hyper-threaded quad core. The performance benefit of additional
+            // threads seems to level off at around six, so we cap it there on
+            // many-core machines (see bug 1431285 comment 14).
+            _ => cmp::min(cmp::max(num_cpus::get() * 3 / 4, 1), 6),
         };
 
         // If num_threads is one, there's no point in creating a thread pool, so
